File: _fix_enviar_wa.js
const fs = require('fs');

const path = "c:\\Users\\jhose\\OneDrive\\Desktop\\PROYECTO DE CODIGO\\LEGADO_BIBLICO_PROD\\index.html";
let text = fs.readFileSync(path, 'latin1'); // Read in latin1 (ANSI) to preserve bytes exactly as they are

// We apply the replacements

// 1. Fix the double-added accidental bad text
const toRemove = `//                                            \r\n        //   VERS CULOS DE ALIENTO  Acceso directo\r\n        //                                            `;
text = text.replace(toRemove, "");
text = text.replace(
    "Cambiar a la pestaa ALIENTO automticamente",
    "Cambiar a la pesta\xF1a ALIENTO autom\xE1ticamente" // pestaña, automáticamente in latin1 map
);
text = text.replace(
    "Restaurar la pantalla que corresponde segn donde estaba",
    "Restaurar la pantalla que corresponde seg\xFAn donde estaba" // según
);

// 2. Add function and update button
const closingBrace = `        function cerrarDevocionalOverlay() {\r
            const ov = document.getElementById('devocional-overlay');\r
            if (ov) ov.remove();\r
        }`;

const newFuncStr = `        function cerrarDevocionalOverlay() {\r
            const ov = document.getElementById('devocional-overlay');\r
            if (ov) ov.remove();\r
        }\r
\r
        // \uD83D\uDCAC COMPARTIR DEVOCIONAL - WHATSAPP\r
        function _compartirSoloTextoClick() {\r
            const icono    = document.getElementById('dev-icon')?.textContent || '\uD83D\uDCD6';\r
            const titulo   = document.getElementById('dev-label')?.textContent || 'Devocional del D\xEDa';\r
            const vers     = document.getElementById('dev-versiculo')?.textContent || '';\r
            const ref      = document.getElementById('dev-referencia')?.textContent || '';\r
            const reflex   = document.getElementById('dev-reflexion')?.textContent || '';\r
\r
            const refLimpia = ref.replace('\uD83D\uDCD6', '').trim();\r
            const msg = \`\${icono} *DEVOCIONAL DEL D\xCDA \u2014 LEGADO B\xCDBLICO*\\n*\${titulo}*\\n\\n\uD83D\uDCD6 \${vers}\\n_\${refLimpia}_\\n\\n\uD83D\uDCAD *Reflexi\xF3n:*\\n_\${reflex}_\\n\\n\uD83D\uDCF2 _Compartido desde Legado B\xEDblico App_\`;\r
\r
            const isMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);\r
            const waUrl = isMobile \r
                ? \`whatsapp://send?text=\${encodeURIComponent(msg)}\` \r
                : \`https://web.whatsapp.com/send?text=\${encodeURIComponent(msg)}\`;\r
            \r
            window.open(waUrl, '_blank');\r
        }`;
text = text.replace(closingBrace, newFuncStr);

const oldBtnRegex = /<button onclick="_compartirSoloTextoClick\(\)" style="[\s\S]*?">.*?Enviar solo como texto.*?<\/button>/;
const newBtnStr = `<button onclick="_compartirSoloTextoClick()" style="
                            width:100%;padding:14px;
                            background:linear-gradient(135deg,#25D366,#128C7E);
                            border:none;border-radius:14px;color:#fff;
                            font-weight:900;font-size:0.95rem;cursor:pointer;
                            box-shadow:0 6px 20px rgba(37,211,102,0.3);
                        ">\uD83D\uDCAC ENVIAR POR WHATSAPP (TEXTO)</button>`;

text = text.replace(oldBtnRegex, newBtnStr);

fs.writeFileSync(path, text, 'latin1');
console.log("Fixes applied successfully.");

File: _BASURA/FIXES/_fix_enviar_wa.py
import codecs
import re

path = r'c:\Users\jhose\OneDrive\Desktop\PROYECTO DE CODIGO\LEGADO_BIBLICO_PROD\index.html'
try:
    with codecs.open(path, 'r', 'utf-8') as f:
        text = f.read()
except UnicodeDecodeError:
    with codecs.open(path, 'r', 'latin-1') as f:
        text = f.read()

# Fix the broken text from earlier replacement if it exists
text = text.replace('Restaurar la pantalla que corresponde segn', 'Restaurar la pantalla que corresponde según')
text = text.replace('//   VERS CULOS DE ALIENTO  Acceso directo', '// 🌟 VERSÍCULOS DE ALIENTO — Acceso directo')

# Add the function _compartirSoloTextoClick()
func_target = """        function cerrarDevocionalOverlay() {
            const ov = document.getElementById('devocional-overlay');
            if (ov) ov.remove();
        }"""

new_func = """        function cerrarDevocionalOverlay() {
            const ov = document.getElementById('devocional-overlay');
            if (ov) ov.remove();
        }

        // 💬 COMPARTIR DEVOCIONAL - WHATSAPP
        function _compartirSoloTextoClick() {
            const icono    = document.getElementById('dev-icon')?.textContent || '📖';
            const titulo   = document.getElementById('dev-label')?.textContent || 'Devocional del Día';
            const vers     = document.getElementById('dev-versiculo')?.textContent || '';
            const ref      = document.getElementById('dev-referencia')?.textContent || '';
            const reflex   = document.getElementById('dev-reflexion')?.textContent || '';

            const refLimpia = ref.replace('📖', '').trim();
            const msg = `${icono} *DEVOCIONAL DEL DÍA — LEGADO BÍBLICO*\\n*${titulo}*\\n\\n📖 ${vers}\\n_${refLimpia}_\\n\\n💭 *Reflexión:*\\n_${reflex}_\\n\\n📲 _Compartido desde Legado Bíblico App_`;

            const isMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
            const waUrl = isMobile 
                ? `whatsapp://send?text=${encodeURIComponent(msg)}` 
                : `https://web.whatsapp.com/send?text=${encodeURIComponent(msg)}`;
            
            window.open(waUrl, '_blank');
        }"""

if func_target in text:
    text = text.replace(func_target, new_func)
    print('Function injected successfully')
else:
    print('Function injection FAILED - target not found')

# Update the button
btn_regex = r'<button onclick=\"_compartirSoloTextoClick\(\)\" style=\"[\s\S]*?\">💬 Enviar solo como texto \(Clásico\)</button>'
new_btn = r'''<button onclick="_compartirSoloTextoClick()" style="
                            width:100%;padding:14px;
                            background:linear-gradient(135deg,#25D366,#128C7E);
                            border:none;border-radius:14px;color:#fff;
                            font-weight:900;font-size:0.95rem;cursor:pointer;
                            box-shadow:0 6px 20px rgba(37,211,102,0.3);
                        ">💬 ENVIAR POR WHATSAPP</button>'''

if re.search(btn_regex, text):
    text = re.sub(btn_regex, new_btn, text)
    print('Button replaced successfully')
else:
    print('Button target not found')

# Save back
# We know the original HTML should be utf-8 or latin-1. Let's force utf-8 so emojis render well natively.
with codecs.open(path, 'w', 'utf-8-sig') as f:
    f.write(text)
print('File saved')
